src/views/ProjectView.vue

```diff
@@ -1,9 +1,127 @@
-    import ImportButton from '../components/ImportButton.vue'
+import { ref } from 'vue'
+import { useRouter } from 'vue-router'
+import { useStore } from 'vuex'
+import { fetchApiCall } from '../utils/api'
+import ToggleButton from '../components/ToggleButton.vue'
+
+const router = useRouter()
+const store = useStore()
+const isPopupVisible = ref(false)
+const uniqueDataSessions = ref([])
+const newSessionName = ref('')
+const errorMessage = ref('')
+
+const getDataSessions = async () => {
+  try {
+    const url = 'http://127.0.0.1:8000/api/datasessions/'
+    const data = await fetchApiCall(url, 'GET')
+    const results = data.results
+    const uniqueNames = new Set()
+
+    uniqueDataSessions.value = results
+      .filter(session => {
+        const isUnique = !uniqueNames.has(session.name) && !uniqueNames.has(session.id)
+        uniqueNames.add(session.name)
+        uniqueNames.add(session.id)
+        return isUnique
+      })
+      .map(session => ({ id: session.id, name: session.name }))
+
+    isPopupVisible.value = true
+  } catch (error) {
+    console.error('Error getting data sessions: ', error)
+  }
+}
+
+const addImagesToExistingSession = async (session) => {
+  try {
+    const selectedImages = store.state.selectedImages
+    const inputData = selectedImages.map(image => ({
+      'source': image.image,
+      'basename': image.basefile_name
+    }))
+
+    const requestBody = {
+      'name': session.name,
+      'input_data': inputData
+    }
+
+    const url = 'http://127.0.0.1:8000/api/datasessions/' + session.id + '/'
+    const data = await fetchApiCall(url, 'PATCH', requestBody)
+
+  } catch (error) {
+    console.error('Error importing images:', error)
+  }
+}
+
+const selectDataSession = (session) => {
+  isPopupVisible.value = false
+  addImagesToExistingSession(session)
+  router.push({ name: 'DataSessions' })
+}
+
+const createNewDataSession = async () => {
+    if (doesSessionNameExist(newSessionName.value)) {
+        errorMessage.value = 'Data Session name already exists. Please choose a different name.'
+        return
+    }
+
+    try {
+        const selectedImages = store.state.selectedImages
+        const inputData = selectedImages.map(image => ({
+            'source': image.image,
+            'basename': image.basefile_name
+        }))
+        const requestBody = { 
+            'name': newSessionName.value,
+            'input_data': inputData 
+        }
+        const url = 'http://127.0.0.1:8000/api/datasessions/'
+        const data = await fetchApiCall(url, 'POST', requestBody)
+
+        isPopupVisible.value = false
+        newSessionName.value = ''
+        errorMessage.value = ''
+
+        router.push({ name: 'DataSessions' })
+    } catch (error) {
+        console.log('Error creating new data session:', error)
+        errorMessage.value = 'Error creating new data session'
+    }
+}
+
+const doesSessionNameExist = (name) => {
```

Comment:
Minor, but `doesSessionNameExist` implies to me a 'yes/no' return value - what if we called it `sessionNameExists` so that it reflects that it's a boolean return? Naming things is hard

Reply:
yes